core/src/main/java/tech/pegasys/web3signer/core/Eth2Runner.java

```diff
@@ -108,21 +120,58 @@ private void registerEth2Routes(
 
   private ArtifactSignerProvider loadSigners(
       final Config config, final Vertx vertx, final MetricsSystem metricsSystem) {
-    final HashicorpConnectionFactory hashicorpConnectionFactory =
-        new HashicorpConnectionFactory(vertx);
-
-    final AbstractArtifactSignerFactory artifactSignerFactory =
-        new BlsArtifactSignerFactory(
-            config.getKeyConfigPath(),
-            metricsSystem,
-            hashicorpConnectionFactory,
-            BlsArtifactSigner::new);
-
-    final Collection<ArtifactSigner> signers =
-        SignerLoader.load(
-            config.getKeyConfigPath(),
-            "yaml",
-            new YamlSignerParser(List.of(artifactSignerFactory)));
+
+    final List<ArtifactSigner> signers = Lists.newArrayList();
+    if (config.getKeyConfigPath() != null) {
+      final HashicorpConnectionFactory hashicorpConnectionFactory =
+          new HashicorpConnectionFactory(vertx);
+
+      final AbstractArtifactSignerFactory artifactSignerFactory =
+          new BlsArtifactSignerFactory(
+              config.getKeyConfigPath(),
+              metricsSystem,
+              hashicorpConnectionFactory,
+              BlsArtifactSigner::new);
+
+      signers.addAll(
+          SignerLoader.load(
+              config.getKeyConfigPath(),
+              "yaml",
+              new YamlSignerParser(List.of(artifactSignerFactory))));
+    }
+
+    if (config.getAzureKeyVaultParameters() != null) {
+      final AzureKeyVaultParameters params = config.getAzureKeyVaultParameters();
+      final AzureKeyVault keyVault =
+          new AzureKeyVault(
+              params.getClientlId(),
+              params.getClientSecret(),
+              params.getTenantId(),
+              params.getKeyVaultName());
+      signers.addAll(
+          keyVault
+              .getAvailableSecrets()
+              .parallelStream()
+              .map(
+                  secretName -> {
+                    final Optional<String> secret = keyVault.fetchSecret(secretName);
+                    if (secret.isPresent()) {
+                      try {
+                        final Bytes privateKeyBytes = Bytes.fromHexString(secret.get());
+                        final BLSKeyPair keyPair =
+                            new BLSKeyPair(BLSSecretKey.fromBytes(Bytes32.wrap(privateKeyBytes)));
+                        return new BlsArtifactSigner(keyPair);
+                      } catch (final Exception e) {
+                        LOG.error(
+                            "Failed to load secret named {} from azure key vault.", secretName);
+                        return null;
+                      }
+                    }
+                    return null;
+                  })
+              .filter(Objects::nonNull)
+              .collect(Collectors.toList()));
```

Comment:
Can the azure signer logic be split out into a seperate function

Reply:
done